main.tf

```
@@ -98,6 +87,38 @@ resource "aws_s3_bucket" "quortex" {
   }
 }
 
+resource "aws_s3_bucket_acl" "quortex" {
+  for_each = var.buckets
+
+  bucket = aws_s3_bucket.quortex[each.value].id
+  acl    = "private"
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "quortex" {
+  for_each = var.expiration != null ? var.buckets : toset([])
```

Something like this would keep the `var.expiration.enabled`
```suggestion
  for_each = var.expiration != null && var.expiration.enabled ? var.buckets : toset([])
```